operator/src/main/java/org/bf2/operator/operands/Canary.java

```diff
@@ -306,15 +306,20 @@ private List<EnvVar> buildEnvVar(ManagedKafka managedKafka, Deployment current)
         String kafkaVersion = managedKafka.getSpec().getVersions().getKafka();
         // takes the current Kafka version if the canary already exists. During Kafka upgrades it doesn't have to change, as any other clients.
         if (current != null) {
-            Optional<EnvVar> kafkaVersionEnvVar = current.getSpec().getTemplate().getSpec().getContainers().stream()
+            Optional<Container> kafkaVersionContainer = current.getSpec().getTemplate().getSpec().getContainers().stream()
```

Comment:
Flat-mapping the stream to the container Envs avoids the intermediate `Optional` in the original code.
```java
Optional<EnvVar> kafkaVersionEnvVar = current.getSpec().getTemplate().getSpec().getContainers().stream()
        .filter(container -> "canary".equals(container.getName()))
        .map(Container::getEnv)
        .flatMap(Collection::stream)
        .filter(ev -> "KAFKA_VERSION".equals(ev.getName()))
        .findFirst();
if (kafkaVersionEnvVar.isPresent()) {
    kafkaVersion = kafkaVersionEnvVar.get().getValue();
}
```